### PR TITLE
Update 'Find closes' calendar event in CycleTimetable

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

Find closes at the end of the day (11:59pm) on 4th October in the 2021 cycle

## Changes proposed in this pull request

Make it so.

## Guidance to review

It should just work ™️ 

Please also review similar change for Find - https://github.com/DFE-Digital/find-teacher-training/pull/886

## Link to Trello card

https://trello.com/c/y2CZ1eMw/3856-eoc-dev-update-find-closed-date-in-timeline-across-find-apply

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by TobyRet** [comment]
The is one failing test for `set_reject_by_default` https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/set_reject_by_default.rb#L36

I'm not sure why we are setting the time to end of day - can anyone clarify?

@JR-G @stevehook 🤞

**Comment by JR-G** [comment]
> The is one failing test for `set_reject_by_default` https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/set_reject_by_default.rb#L36
> 
> I'm not sure why we are setting the time to end of day - can anyone clarify?
> 
> @JR-G @stevehook 🤞

@TobyRet EoD was set so the candidate will have the full day before the deadline date, before their application gets RBD

**Comment by TobyRet** [comment]
> > The is one failing test for `set_reject_by_default` https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/set_reject_by_default.rb#L36
> > I'm not sure why we are setting the time to end of day - can anyone clarify?
> > @JR-G @stevehook 🤞
> 
> @TobyRet EoD was set so the candidate will have the full day before the deadline date, before their application gets RBD

@JR-G 

🤔 This does not seem right to me - RBD is later than the deadline

```
[5] pry(main)> CycleTimetable.find_closes
  SiteSetting Load (0.3ms)  SELECT "site_settings".* FROM "site_settings" WHERE "site_settings"."name" = $1 LIMIT $2  [["name", "cycle_schedule"], ["LIMIT", 1]]
=> Mon, 04 Oct 2021 00:00:00.000000000 BST +01:00
[6] pry(main)> 0.business_days.before(CycleTimetable.find_closes).end_of_day
  SiteSetting Load (0.4ms)  SELECT "site_settings".* FROM "site_settings" WHERE "site_settings"."name" = $1 LIMIT $2  [["name", "cycle_schedule"], ["LIMIT", 1]]
=> Mon, 04 Oct 2021 23:59:59.999999999 BST +01:00
```

As the deadline falls at 12AM Monday, should we be doing this instead?

```
[7] pry(main)> 1.business_days.before(CycleTimetable.find_closes).end_of_day
  SiteSetting Load (0.4ms)  SELECT "site_settings".* FROM "site_settings" WHERE "site_settings"."name" = $1 LIMIT $2  [["name", "cycle_schedule"], ["LIMIT", 1]]
=> Fri, 01 Oct 2021 23:59:59.999999999 BST +01:00
```

**Comment by JR-G** [comment]
> > > The is one failing test for `set_reject_by_default` https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/set_reject_by_default.rb#L36
> > > I'm not sure why we are setting the time to end of day - can anyone clarify?
> > > @JR-G @stevehook 🤞
> > 
> > 
> > @TobyRet EoD was set so the candidate will have the full day before the deadline date, before their application gets RBD
> 
> @JR-G
> 
> 🤔 This does not seem right to me - RBD is later than the deadline
> 
> ```
> [5] pry(main)> CycleTimetable.find_closes
>   SiteSetting Load (0.3ms)  SELECT "site_settings".* FROM "site_settings" WHERE "site_settings"."name" = $1 LIMIT $2  [["name", "cycle_schedule"], ["LIMIT", 1]]
> => Mon, 04 Oct 2021 00:00:00.000000000 BST +01:00
> [6] pry(main)> 0.business_days.before(CycleTimetable.find_closes).end_of_day
>   SiteSetting Load (0.4ms)  SELECT "site_settings".* FROM "site_settings" WHERE "site_settings"."name" = $1 LIMIT $2  [["name", "cycle_schedule"], ["LIMIT", 1]]
> => Mon, 04 Oct 2021 23:59:59.999999999 BST +01:00
> ```
> 
> Given Find closes at 12AM Monday, should we be doing this instead?
> 
> ```
> [7] pry(main)> 1.business_days.before(CycleTimetable.find_closes).end_of_day
>   SiteSetting Load (0.4ms)  SELECT "site_settings".* FROM "site_settings" WHERE "site_settings"."name" = $1 LIMIT $2  [["name", "cycle_schedule"], ["LIMIT", 1]]
> => Fri, 01 Oct 2021 23:59:59.999999999 BST +01:00
> ```

Yeah... that is odd... `1.business_days.before` didn't work, it was returning 2 days before, which is why we went with `0.business_days` instead 🤔

**Comment by TobyRet** [comment]
@JR-G @duncanjbrown 
Are you happy with this change? https://github.com/DFE-Digital/apply-for-teacher-training/pull/5397/commits/9847b250c1223c98a5fadaef79324d8c380f19a3

**Review by stevehook — APPROVED** [review]
This looks good assuming find closes at 0:00. If say you had:

```
find_closes: Time.zone.local(2021, 10, 4, 17, 30, 0)
```

then

```
1.business_days.before(CycleTimetable.find_closes).end_of_day -> Fri, 01 Oct 2021 23:59:59
```

I _think_ that makes complete sense but if:

```
find_closes: Time.zone.local(2021, 10, 4, 17, 23, 59, 59)
```

we might need to revisit.

**Comment by JR-G** [comment]
> @JR-G @duncanjbrown
> Are you happy with this change? [9847b25](https://github.com/DFE-Digital/apply-for-teacher-training/commit/9847b250c1223c98a5fadaef79324d8c380f19a3)

Yep, tested it locally and it's returning the correct date & time

File: app/services/cycle_timetable.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ class CycleTimetable
       show_deadline_banner: Time.zone.local(2021, 8, 1, 9),
       apply_1_deadline: Time.zone.local(2021, 9, 7, 18),
       apply_2_deadline: Time.zone.local(2021, 9, 21, 18),
-      find_closes: Time.zone.local(2021, 10, 3),
+      find_closes: Time.zone.local(2021, 10, 4, 23, 59, 59),
```
Comment (duncanjbrown) [inline app/services/cycle_timetable.rb:26]: Could we give times to all the other `find_closes` dates while we're here?

**Pull request merged**